From 30241086e9c59e5e0d90797adaf1cdb7a14fa620 Mon Sep 17 00:00:00 2001
From: Eli Nidam <elini@marvell.com>
Date: Sun, 29 Dec 2013 15:30:12 +0200
Subject: [PATCH 1253/1825] a38x: support NOR module

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit b6163db59f1dea1757a83701ffa273c66df64cdc

	1. Added board initialize for NOR module
	2. Update MPP for NOR module
	3. Enable SPI1 when NOR module detect.

Change-Id: If8a9af0941767786c81df1b1732f805ec2091c9a
Signed-off-by: Eli Nidam <elini@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/4946
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Nadav Haklai <nadavh@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../armada_38x_family/boardEnv/mvBoardEnvLib.h     |    2 +-
 .../armada_38x_family/boardEnv/mvBoardEnvSpec.c    |   14 +++++++++-----
 2 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.h b/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.h
index 77bae7b..ff78281 100644
--- a/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.h
+++ b/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvLib.h
@@ -405,7 +405,7 @@ struct _mvBoardMppModule {
 #define MPP_TDM_MODULE		{ {6, 0x45333333}, {7, 0x00004444} }
 #define MPP_I2S_MODULE		{6, 0x55544554}
 #define MPP_SPDIF_MODULE	{6, 0x55444444}
-#define MPP_NOR_MODULE		{ {0, 0x51111111}, {1, 0x11555555}, {2, 0x55566011}, \
+#define MPP_NOR_MODULE		{ {0, 0x55111111}, {1, 0x15555555}, {2, 0x55566011}, \
 				  {3, 0x55555055}, {4, 0x55555555}, {5, 0x40045525 } }
 #define MPP_NAND_MODULE		{ {0, 0x55111111}, {1, 0x15555555}, {2, 0x55266011}, \
 				  {3, 0x25255051}, {4, 0x25555555}, {5, 0x40042555 } }
diff --git a/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvSpec.c b/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvSpec.c
index fa8d22a..46789c0 100644
--- a/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvSpec.c
+++ b/arch/arm/mach-armada38x/armada_38x_family/boardEnv/mvBoardEnvSpec.c
@@ -114,12 +114,15 @@ MV_BOARD_MAC_INFO db88f68xxInfoBoardMacInfo[] = {
 
 MV_DEV_CS_INFO db88f68xxInfoBoardDeCsInfo[] = {
 	/*{deviceCS, params, devType, devWidth, busWidth }*/
-#if defined(MV_INCLUDE_SPI)
-	{ SPI0_CS0,	N_A, BOARD_DEV_SPI_FLASH,	8,	8 } /* SPI DEV */
-#endif
 #ifdef MV_NAND
 	{ DEVICE_CS0,	N_A, BOARD_DEV_NAND_FLASH,	8,	8},  /* NAND DEV */
 #endif
+#if defined(MV_INCLUDE_NOR)
+	{ DEV_BOOCS,	N_A, BOARD_DEV_NOR_FLASH,	16,	16}, /* NOR DEV */
+	{ SPI0_CS1,	N_A, BOARD_DEV_SPI_FLASH,	8,	8 } /* SPI DEV */
+#else
+	{ SPI0_CS0,	N_A, BOARD_DEV_SPI_FLASH,	8,	8 } /* SPI DEV */
+#endif
 };
 /*
 MV_BOARD_MPP_TYPE_INFO db88f68XXInfoBoardModTypeInfo[] = {
@@ -183,8 +186,9 @@ MV_BOARD_INFO db88f68xx_board_info = {
 	.pBoardTdmSpiInfo		= db88f68xxTdmSpiInfo,
 
 	/* NOR init params */
-	.norFlashReadParams		= 0,
-	.norFlashWriteParams		= 0,
+	.norFlashReadParams		= DB_88F68XX_BOARD_NOR_READ_PARAMS,
+	.norFlashWriteParams		= DB_88F68XX_BOARD_NOR_WRITE_PARAMS,
+
 	/* Enable modules auto-detection. */
 	.configAutoDetect		= MV_TRUE,
 	.numIoExp			= 0,
-- 
1.7.5.4

